src/PIL/GimpPaletteFile.py

```diff
@@ -24,32 +25,44 @@ class GimpPaletteFile:
 
     rawmode = "RGB"
 
-    def __init__(self, fp):
+    #: override if reading larger palettes is needed
+    max_colors = 256
+    _max_line_size = 100
+    _max_file_size = 2**20  # 1MB
 
-        self.palette = [o8(i) * 3 for i in range(256)]
+    def __init__(self, fp):
 
         if fp.readline()[:12] != b"GIMP Palette":
             raise SyntaxError("not a GIMP palette file")
 
-        for i in range(256):
+        read = 0
+
+        self.palette = b""
+        while len(self.palette) < 3 * self.max_colors:
 
-            s = fp.readline()
+            s = fp.readline(self._max_file_size)
```

Comment:
```suggestion
            s = fp.readline(self._max_line_size)
```

Reply:
This looks like a mistake, where you've accidentally used the file size limit for the line size limit for [the size argument.](https://docs.python.org/3/library/io.html#io.IOBase.readline)

Reply:
no - "file" size is deliberate here - to avoid DoS only. 
line size checks are made just bellow, and as far as I know it is something just PIL has - there is no spec for this kind of file, but for GIMP source code, and there is no line size limit there.

Changing here for reading linesize would make te file break with long comments (as it did), and that makes no sense:

if one has a python project in which Palttes with this format are used, there is no reason to impose an arbitrary limit on comments on that source code.   Resilience against malicious files when those are input by 3rdy parties are mitigated through the file-size check.